### PR TITLE
[sc-16014] Use backend instead of cluster to generate DS api key

### PR DESCRIPTION
**Pull request by shachakz**

### CODE REVIEW

File: cmd/login.go [hunk 1]
```diff
@@ -141,26 +141,26 @@ func selectClusterName(tenant *api.TenantInfo) (string, error) {
...
+	backendId := ""
 
-	switch len(clusterNames) {
+	switch len(backendNames) {
 	case 0:
-		return "", errors.New("no active clusters")
+		return "", errors.New("no active backends")
 	case 1:
-		clusterId = clusterNames[0]
+		backendId = backendNames[0]
```
Comment (ErezLevip) [inline cmd/login.go:160]: just pick a convention here, id/name

File: cmd/deploy.go [hunk 2]
```diff
@@ -492,7 +492,7 @@ func validateClusterRegistered(ctx context.Context, tenantUUID, clusterName stri
 
 	apiClient := api.NewClient(auth0Token)
 
-	if err = apiClient.PollIsClusterExist(ctx, tenantUUID, clusterName); err != nil {
+	if err = apiClient.PollIsBackendExist(ctx, tenantUUID, clusterName); err != nil {
```
Comment (ErezLevip) [inline cmd/deploy.go:495]: why is this still a cluster?

Comment (ErezLevip) [reply cmd/deploy.go:495]: what will happen if i install a cluster thats connected to an incloud deployment where cluster and backend name are not the same?

**Review by ErezLevip — CHANGES_REQUESTED** [review]

File: cmd/deploy.go (hunk 2 above)
Comment (shachakz) [reply cmd/deploy.go]: I changed it to poll on cluster, without using backend at all..
In inCluster, users do not provide backendName (it's not part of the installation flow) so currently we can't validate installation using the backend name and we need to re-think on it.

For incloud, this flow is irrelevant - we install directly with helm charts

**Review by ErezLevip — APPROVED** [review]

**Pull request merged**